controllers/rhmi/bootstrapReconciler.go

```diff
@@ -762,6 +770,43 @@ func (r *Reconciler) processQuota(installation *rhmiv1alpha1.RHMI, namespace str
 	return nil
 }
 
+func (r *Reconciler) reconcileCustomSMTP(ctx context.Context, serverClient k8sclient.Client) (rhmiv1alpha1.StatusPhase, error) {
+
+	smtp, err := cs.GetCustomAddonValues(serverClient, r.installation.Namespace)
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, err
+	}
+
+	validation := cs.ParameterValidation(smtp)
+	if validation == cs.Valid {
+		phase, err := cs.CreateOrUpdateCustomSMTPSecret(ctx, serverClient, smtp, r.installation.Namespace)
+		if err != nil {
+			return phase, err
+		}
+		r.installation.Status.CustomSmtp = &rhmiv1alpha1.CustomSmtpStatus{}
+		r.installation.Status.CustomSmtp.Active = true
+
+	} else if validation == cs.Partial {
+		phase, err := cs.DeleteCustomSMTP(ctx, serverClient, r.installation.Namespace)
+		if err != nil {
+			return phase, err
+		}
+
+		errorString := cs.ParameterErrors(smtp)
+		r.installation.Status.CustomSmtp.Active = false
+		r.installation.Status.CustomSmtp.Error = fmt.Sprintf("Custom SMTP partial configured, missing fields: %s", errorString)
+
+	} else {
+		phase, err := cs.DeleteCustomSMTP(ctx, serverClient, r.installation.Namespace)
+		if err != nil {
```

Comment:
I guess form a safety point we can set the value to false if there is an error given when trying to delete the custom smtp secret.

Reply:
Sorry, I put my comment in the wrong place. I meant in that else {} block it should be set to false to cover the scenario where it is currently true but the customer has deleted all values in ocm, in order to default back to the RHOAM supplied SMTP creds. @Boomatang

Reply:
When the customer removes all the values from ocm the custom smtp status block gets zero out in the rhmi cr. As it will not exist there is no need to set it to false.